File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummy = new ListNode();
        ListNode curr = dummy;
        int v1 = 0;
        int v2 = 0;
        int carry = 0;

        while (l1 != null || l2 != null || carry != 0) {
            if (l1 != null) {
                v1 = l1.val;
            } else {
                v1 = 0;
            }

            if (l2 != null) {
                v2 = l2.val;
            } else {
                v2 = 0;
            }
            int add = v1 + v2 + carry;
            carry = add / 10;
            add = add % 10;

            ListNode nn = new ListNode(add);
            curr.next = nn;

            curr = curr.next;

            if (l1 != null) {
                l1 = l1.next;
            } else {
                l1 = null;
            }

            if (l2 != null) {
                l2 = l2.next;
            } else {
                l2 = null;
            }
        }

        return dummy.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* dummy = new ListNode(0);
        ListNode* curr = dummy;
        int carry = 0;
        int val1 = 0;
        int val2 = 0;
        
        while(l1!=NULL || l2!=NULL || carry!=0){
            
            //Extract L1
            if(l1!=NULL){
                val1 = l1->val;
            } else {
                val1 = 0;
            }
            
            //Extract l2
            if(l2!=NULL){
                val2 = l2->val;
            } else{
                val2 = 0;
            }
            
            int val = val1 + val2 + carry;
            carry = val/10;
            int numadd = val%10;
            
            curr->next = new ListNode(numadd);
            curr = curr->next;
            
            //Extract L1
            if(l1!=NULL){
                l1 = l1->next;
            } else {
                l1 = NULL;
            }
            
            //Extract l2
            if(l2!=NULL){
                l2 = l2->next;
            } else{
                l2 = NULL;
            }
            
        }
        
        return dummy->next;
    }
};